modules/apps/portal-search/portal-search/src/main/java/com/liferay/portal/search/internal/spi/model/query/contributor/GroupIdQueryPreFilterContributor.java

```diff
@@ -98,16 +100,20 @@ public void contribute(
 	private void _addInactiveGroupsBooleanFilter(
 		BooleanFilter booleanFilter, SearchContext searchContext) {
 
+		long companyId = searchContext.getCompanyId();
+
 		List<Long> inactiveGroupIds = _groupLocalService.getGroupIds(
-			searchContext.getCompanyId(), false);
+			companyId, false);
 
 		if (ListUtil.isEmpty(inactiveGroupIds)) {
 			return;
 		}
 
 		TermsFilter groupIdTermsFilter = new TermsFilter(Field.GROUP_ID);
 
-		int maxTermsCount = 65536; //Could be set by a config
+		int maxTermsCount = GetterUtil.getInteger(
```

Comment:
We don't want to retrieve the index settings from the search engine every search. We need to come up with a different way. How did we handle index.max_result_window? We could probably do the same.

Reply:
Actually with max result window we were using it in elasticsearch module so we could get it directly from elasticsearchConfigurationWrapper, now this code is in portal-search, so we will need to create a new way to access this information without retrieve from the index. Probably I will create a new interface to get this information.

Reply:
just my opinion, im thinking we just leave it hardcoded as 65536, which is the default, and that should be good enough. https://www.elastic.co/guide/en/elasticsearch/reference/current/index-modules.html#index-max-terms-count

as far as i know, the only reason customers have changed this value in the past was because it was the only "solution" we had to this issue. but i dont think customers would/should know about this setting otherwise, probably better to just keep it under the hood.